internal/app/machined/pkg/system/services/apid.go

```diff
@@ -113,7 +113,17 @@ func (o *APID) Runner(config runtime.Configurator) (runner.Runner, error) {
 
 	env := []string{}
 	for key, val := range config.Machine().Env() {
-		env = append(env, fmt.Sprintf("%s=%s", key, val))
+		switch key {
+		// explicitly exclude proxy variables from apid since this will
+		// negatively impact grpc connections.
+		// ref: https://github.com/grpc/grpc-go/blob/0f32486dd3c9bc29705535bd7e2e43801824cbc4/clientconn.go#L199-L206
+		// ref: https://github.com/grpc/grpc-go/blob/63ae68c9686cc0dd26c4f7476d66bb2f5c31789f/proxy.go#L118-L144
+		case strings.ToLower("no_proxy"):
```

Comment:
should it be `switch strings.ToLower(key)` ?

Reply:
👍

Reply:
worked for me :trollface: